compiler/ast/checked_ast.nim

```diff
@@ -25,10 +29,71 @@ type
   CheckedAst* = object
     ## Represents either in-progress or fully semanticized AST. It is meant
     ## as a wrapper for semantically *checked* AST, hence the name.
-    n: PNode       ## stores the AST. Never nil
+    n: PNode       ## stores the AST. Never nil nor an error
     hasError: bool ## tracks whether `n` contains an ``nkError`` node
                    ## somewhere
 
+  ElaborateAst* = object
+    ## *Note*: the name is not final and going to change.
+    ## Represents an in-progress or finished production of either an AST of the
+    ## same kind as the source, or an error.
```

Comment:
```suggestion
    ## Represents an in-progress or finished production of either an AST of the
    ## same kind and number of children as the source, or an error.
```

Reply:
Yup, being clear here is important -- I changed it.